pylint/utils/pragma_parser.py

```diff
@@ -0,0 +1,145 @@
+# -*- coding: utf-8 -*-
+
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/master/COPYING
+
+import re
+from collections import namedtuple
+from typing import Generator, List
+
+# Allow stopping after the first semicolon/hash encountered,
+# so that an option can be continued with the reasons
+# why it is active or disabled.
+OPTION_RGX = r"""
+    \s*                # Any number of whithespace
+    \#?                # One or zero hash
+    .*                 # Anything (as much as possible)
+    (\s*               # Beginning of first matched group and any number of whitespaces
+    \#                 # Beginning of comment
+    .*?                # Anything (as little as possible)
+    \bpylint:          # pylint word and column
+    \s*                # Any number of whitespaces
+    ([^;#\n]+))        # Anything except semicolon or hash or newline (it is the second matched group) 
+                       # and end of the first matched group
+    [;#]{0,1}"""  # From 0 to 1 repetition of semicolon or hash
+OPTION_PO = re.compile(OPTION_RGX, re.VERBOSE)
+
+
+PragmaRepresenter = namedtuple("PragmaRepresenter", "action messages")
+
+
+ATOMIC_KEYWORDS = ("disable-all", "skip-file")
+MESSAGE_KEYWORDS = ("disable-msg", "enable-msg", "disable", "enable")
```

Comment:
This can be a frozenset:

```suggestion
MESSAGE_KEYWORDS = frozenset(("disable-msg", "enable-msg", "disable", "enable"))
```